app/models/manageiq/providers/amazon/inventory/collector/target_collection.rb

```diff
@@ -140,9 +140,11 @@ def floating_ips
     return [] if references(:floating_ips).blank?
 
     multi_query(references(:floating_ips)) do |refs|
-      hash_collection.new(
-        aws_ec2.client.describe_addresses(:filters => [{:name => 'allocation-id', :values => refs}]).flat_map(&:addresses)
-      ).all
+      results = aws_ec2.client.describe_addresses.flat_map(&:addresses).select { |addr|
+        refs.include?(addr.allocation_id) || refs.include?(addr.public_ip)
```

Comment:
@djberg96 could you do two filters, one for allocation-id and one for public-ip?

Something like
```
aws_ec2.client.describe_addresses(:filters => [{:name => 'allocation-id', :values => refs}]).flat_map(&:addresses) +
  aws_ec2.client.describe_addresses(:filters => [{:name => 'public-ip', :values => refs}]).flat_map(&:addresses)
```

Reply:
Sure could, though now we're making two calls instead of one, though it would potentially be a memory savings. If you prefer that approach, though, I can make it.

Reply:
Yeah just thinking we will only have a hand full of target refs and possibly thousands/tens-of-thousands of total addresses so to bring them all back vs filtering them on the aws side

Reply:
@agrare ok, done